lib/kitchen/driver/ec2.rb

```diff
@@ -98,10 +99,10 @@ class Ec2 < Kitchen::Driver::SSHBase
         end
       end
 
-      def self.iam_creds
+      def iam_creds
         @iam_creds ||= begin
-          fetch_credentials(use_iam_profile:true)
-        rescue RuntimeError => e
+          config[:use_iam_profile] ? fetch_credentials(use_iam_profile: true) : {}
+        rescue RuntimeError, NoMethodError => e
```

Comment:
fetch_credentials calls super when it fails, so since we don't have a super for it to fail over to we need to catch NoMethodError, I think I made a note in the commit message.